File: cliente.ts
const BASE_URL = "http://localhost:8080";
const HEADERS = { "Content-Type": "application/json" };

async function dispararRequisicao(passo: string, endpoint: string, metodo: string, corpo?: any) {
    const opcoes: RequestInit = { method: metodo, headers: HEADERS };
    if (corpo) opcoes.body = JSON.stringify(corpo);

    try {
        const response = await fetch(`${BASE_URL}${endpoint}`, opcoes);
        if (passo !== "Oculto") {
            console.log(`\n--- ${passo} ---`);
            console.log(`Status Code: ${response.status}`);
            const contentType = response.headers.get("content-type");
            if (contentType && contentType.includes("application/json")) {
                const json = await response.json();
                console.log(JSON.stringify(json, null, 2));
            } else {
                console.log("Resposta do Servidor:", await response.text());
            }
        }
    } catch (error) {
        console.error("Erro na conexão:", error);
    }
}

async function testarCenarioTS() {
    console.log("Iniciando cenário TYPESCRIPT...\n");

    // 1. Clientes
    const reqCarlos = { id: 30, nome: "Carlos Mendes", email: "carlos@email.com" };
    const reqAna = { id: 40, nome: "Ana Costa", email: "ana@email.com" };
    
    await dispararRequisicao("Oculto", "/clientes", "POST", reqCarlos);
    await dispararRequisicao("Oculto", "/clientes", "POST", reqAna);
    console.log("Clientes do TS (Carlos e Ana) criados.");

    // 2. Imóvel (Apto pertence à Ana ID 40)
    const reqImovel = {
        id: 102,
        valor: 280000.0,
        tamanho: 85.0,
        endereco: { rua: "Avenida Central", numero: 456, bairro: "Planalto", cidade: "Quixadá" },
        proprietario: { id: 40 }
    };
    await dispararRequisicao("Oculto", "/imoveis", "POST", reqImovel);
    console.log("Imóvel do TS (ID 102) criado.");

    // 3. Contrato (Carlos ID 30 aluga o Apto ID 102)
    const reqContrato = {
        id: 2,
        valorMensal: 1200.0,
        locatario: { id: 30 },
        propriedade: { id: 102 }
    };
    
    await dispararRequisicao("Contrato do TS (ID 2) finalizado", "/contratos", "POST", reqContrato);
    await dispararRequisicao("Consultando TODOS os Contratos da Memória", "/contratos", "GET");
}

testarCenarioTS();

File: cliente.py
import requests
import json

BASE_URL = "http://localhost:8080"
HEADERS = {"Content-Type": "application/json"}

def imprimir_resposta(passo, response):
    print(f"\n--- {passo} ---")
    print(f"Status Code: {response.status_code}")
    try:
        print(json.dumps(response.json(), indent=2, ensure_ascii=False))
    except:
        print("Resposta do Servidor:", response.text)

def testar_cenario_python():
    print("Iniciando cenário PYTHON...")

    # 1. Clientes
    req_joao = {"id": 10, "nome": "João Silva", "email": "joao@email.com"}
    req_maria = {"id": 20, "nome": "Maria Oliveira", "email": "maria@email.com"}
    
    requests.post(f"{BASE_URL}/clientes", json=req_joao, headers=HEADERS)
    requests.post(f"{BASE_URL}/clientes", json=req_maria, headers=HEADERS)
    print("Clientes do Python (João e Maria) criados.")

    # 2. Imóvel (Casa pertence ao João ID 10)
    req_imovel = {
        "id": 101,
        "valor": 350000.0,
        "tamanho": 120.5,
        "endereco": { "rua": "Rua das Flores", "numero": 123, "bairro": "Centro", "cidade": "Quixadá" },
        "proprietario": { "id": 10 } 
    }
    requests.post(f"{BASE_URL}/imoveis", json=req_imovel, headers=HEADERS)
    print("Imóvel do Python (ID 101) criado.")

    # 3. Contrato (Maria ID 20 aluga a Casa ID 101)
    req_contrato = {
        "id": 1,
        "valorMensal": 1500.0,
        "locatario": { "id": 20 },
        "propriedade": { "id": 101 }
    }
    res_contrato = requests.post(f"{BASE_URL}/contratos", json=req_contrato, headers=HEADERS)
    imprimir_resposta("Contrato do Python (ID 1) finalizado", res_contrato)
    
    res_get = requests.get(f"{BASE_URL}/contratos")
    imprimir_resposta("Consultando TODOS os Contratos da Memória", res_get)

if __name__ == "__main__":
    testar_cenario_python()